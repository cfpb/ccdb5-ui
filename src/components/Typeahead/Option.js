// Adapted from https://github.com/fmoo/react-typeahead

import PropTypes from 'prop-types';
import React from 'react';

export default class Option extends React.Component {
  render() {
    const classes = ['typeahead-option', 'body-copy'];
    if (this.props.isSelected) {
      classes.push('selected');
    }

    return (
      <li
        className={classes.join(' ')}
        onMouseDown={this.props.onClick}
        onTouchStart={this.props.onClick}
      >
        {this.props.children}
      </li>
    );
  }
}

Option.propTypes = {
  children: PropTypes.node,
  isSelected: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
};

Option.defaultProps = {
  isSelected: false,
};
