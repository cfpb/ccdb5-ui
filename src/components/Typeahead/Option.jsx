// Adapted from https://github.com/fmoo/react-typeahead

import PropTypes from 'prop-types';
import React from 'react';

export default class Option extends React.Component {
  render() {
    const classes = [ 'typeahead-option', 'body-copy' ];
    if ( this.props.selected ) {
      classes.push( 'selected' );
    }

    return (
      <li className={classes.join( ' ' )}
        onMouseDown={this.props.onClick}
        onTouchStart={this.props.onClick}>
        { this.props.children }
      </li>
    );
  }
}

Option.propTypes = {
  selected: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

Option.defaultProps = {
  selected: false
};
