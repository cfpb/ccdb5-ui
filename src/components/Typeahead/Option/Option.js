// Adapted from https://github.com/fmoo/react-typeahead

import PropTypes from 'prop-types';
import React from 'react';

export const Option = ({ children, isSelected = false, onClick }) => {
  const classes = ['typeahead-option', 'body-copy'];
  if (isSelected) {
    classes.push('selected');
  }

  return (
    <li
      className={classes.join(' ')}
      onMouseDown={onClick}
      onTouchStart={onClick}
    >
      {children}
    </li>
  );
};

Option.propTypes = {
  children: PropTypes.node,
  isSelected: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
};
