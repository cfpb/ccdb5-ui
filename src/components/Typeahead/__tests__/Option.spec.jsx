import React from 'react';
import renderer from 'react-test-renderer';
import Option from '../Option';

describe('component::Option', () => {
  it('looks different when selected', () => {
    const target = renderer.create(
      <Option selected={true} onClick={jest.fn()} />
    );

    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
