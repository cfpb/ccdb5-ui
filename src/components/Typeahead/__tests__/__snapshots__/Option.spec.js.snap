// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::Option looks different when selected 1`] = `
<li
  className="typeahead-option body-copy selected"
  onMouseDown={[MockFunction]}
  onTouchStart={[MockFunction]}
/>
`;
